.github/workflows/codecov-publish.yml

```diff
@@ -0,0 +1,56 @@
+name: Codecov Publish
+
+# Due to the behavior of workflow_run, changes to this file will
+# only be reflected once it is pushed up to the `default` (main) branch
+#
+# It is recommended to test changes to this file by pushing to the default branch
+# on a fork and evaluating the correctness of the action execution
+
+on:
+  workflow_run:
+    workflows: [CI]
+    types:
+      - completed
+
+jobs:
+  publish-codecov:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
```

Comment:
Why only report on success?  IMO code coverage numbers are still useful even if a test failed

Reply:
Currently, we have a lot of dependencies, particularly with gradle tests that would make reporting coverage with failures not working today. E.g. we don't run the `isolatedTest` for a task if the `test` task failed.

I don't want to add this without it being accurate, so let's track it separately.